### PR TITLE
Move CI to GitHub Actions

### PR DESCRIPTION
**Pull request by jcs090218**
This is implementation of issue #244. 

@JesusMtnez Can you please take a look? Thanks! :)

### CODE REVIEW

File: dashboard-widgets.el [hunk 1]
```diff
@@ -484,7 +484,9 @@ If MESSAGEBUF is not nil then MSG is also written in message buffer."
                      (when (and (fboundp 'image-transforms-p)
                                 (memq 'scale (funcall 'image-transforms-p)))
                        size-props))))
-           (size (image-size spec))
+           ;; TODO: For some reason, `elisp-lint' is reporting error void
+           ;; function `image-size'.
+           (size (when (fboundp 'image-size) (image-size spec)))
```
Comment (jcs090218) [inline dashboard-widgets.el:489]: I have added the note above. Function `image-size` is reporting void function in macOS and Linux. Not sure why. 😕

File: scripts/test-pkg.bat [hunk 2]
```diff
@@ -0,0 +1,9 @@
+@echo off
+
+echo "Executing package..."
+
+emacs -nw --batch --load "./scripts/test-pkg.el"
+
+git clone "https://github.com/gonewest818/elisp-lint"
+
+emacs --batch -q -l "./scripts/pkg-prepare.el" -l "./elisp-lint/elisp-lint.el" -f elisp-lint-files-batch --no-package-lint dashboard-widgets.el dashboard.el
```
Comment (jcs090218) [inline scripts/test-pkg.bat:9]: Here starts the linting process. (Windows)

File: scripts/test-pkg.el [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  '(page-break-lines org all-the-icons)
+  '(elisp-lint package-lint dash))
+ t)
+
+;;
+;; (@* "Test" )
+;;
+
+(require 'dashboard)
+(dashboard-insert-startupify-lists)
```
Comment (jcs090218) [inline scripts/test-pkg.el:20]: You can more test here.

File: scripts/test-pkg.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
+echo "Executing package..."
+
+${EMACS:=emacs} -nw --batch --load "./scripts/test-pkg.el"
+
+git clone "https://github.com/gonewest818/elisp-lint"
+
+emacs --batch -q -l "./scripts/pkg-prepare.el" -l "./elisp-lint/elisp-lint.el" -f elisp-lint-files-batch --no-package-lint dashboard-widgets.el dashboard.el
```
Comment (jcs090218) [inline scripts/test-pkg.sh:7]: Here starts the linting process. (macOS and Linux)

**Review by jcs090218 — COMMENTED** [review]
The CI passed!

https://github.com/emacs-dashboard/emacs-dashboard/actions/runs/481864438

**Comment by JesusMtnez** [comment]
@jcs090218 Thank you so much for this migration!

One thing we could consider to change (in this PR or another one) is making the check for the latest emacs snapshot optional. What do you think about that?

**Review by JesusMtnez — APPROVED** [review]

**Comment by jcs090218** [comment]
> One thing we could consider to change (in this PR or another one) is making the check for the latest emacs snapshot optional. What do you think about that?

Sounds good to me! ;) I don't know how to do that properly yet. I will make it in the later PR.

**Pull request merged**